crates/solvers-dto/src/auction.rs

```diff
@@ -4,61 +4,104 @@ use {
     bigdecimal::BigDecimal,
     number::serialization::HexOrDecimalU256,
     serde::Deserialize,
+    serde_json::{Number, Value},
     serde_with::{serde_as, DisplayFromStr},
     std::collections::HashMap,
+    utoipa::{
+        openapi::{
+            AllOfBuilder,
+            ArrayBuilder,
+            ObjectBuilder,
+            OneOfBuilder,
+            Ref,
+            RefOr,
+            Schema,
+            SchemaType::{self},
+        },
+        ToSchema,
+    },
     web3::types::{H160, H256, U256},
 };
 
+/// The abstract auction to be solved by the searcher.
 #[serde_as]
-#[derive(Debug, Deserialize)]
+#[derive(Debug, Deserialize, ToSchema)]
 #[serde(rename_all = "camelCase")]
 pub struct Auction {
+    /// An opaque identifier for the auction. Will be set to `null` for requests
+    /// that are not part of an auction (when quoting token prices for example).
     #[serde_as(as = "Option<DisplayFromStr>")]
+    #[schema(value_type = String)]
     pub id: Option<i64>,
-    pub tokens: HashMap<H160, Token>,
+    /// A map of token addresses to token information.
+    pub tokens: HashMap<H160, TokenInfo>,
+    /// The solvable orders included in the auction.
     pub orders: Vec<Order>,
+    /// On-chain liquidity that can be used by the solution.
     pub liquidity: Vec<Liquidity>,
+    /// The current estimated gas price that will be paid when executing a
+    /// settlement. Additionally, this is the gas price that is multiplied with
+    /// a settlement's gas estimate for solution scoring.
     #[serde_as(as = "HexOrDecimalU256")]
+    #[schema(value_type = TokenAmount)]
     pub effective_gas_price: U256,
+    /// The deadline by which a solution to the auction is required. Requests
+    /// that go beyond this deadline are expected to be cancelled by the caller.
+    #[schema(value_type = DateTime)]
     pub deadline: chrono::DateTime<chrono::Utc>,
 }
 
+/// CoW Protocol order information relevant to execution.
 #[serde_as]
-#[derive(Debug, Deserialize)]
+#[derive(Debug, Deserialize, ToSchema)]
 #[serde(rename_all = "camelCase")]
 pub struct Order {
+    #[schema(value_type = OrderUid)]
     #[serde_as(as = "serialize::Hex")]
     pub uid: [u8; 56],
+    #[schema(value_type = Token)]
     pub sell_token: H160,
+    #[schema(value_type = Token)]
     pub buy_token: H160,
+    #[schema(value_type = TokenAmount)]
```

Comment:
what do you think of creating a "Deserializable" type called `TokenAmount` (if it doesn't exist already), derive the proper schema to the new type, and use the type all over the deserializable structs?